### PR TITLE
Add wildcard tolerations for rke addons

### PR DESCRIPTION
**Pull request by orangedeng**

### CODE REVIEW

**Comment by loganhz** [comment]
lgtm

**Comment by loganhz** [comment]
Link to https://github.com/rancher/rancher/issues/13972

**Comment by cjellick** [comment]
Needs rebased

File: rke/templates/calico.go [hunk 1]
```diff
@@ -505,6 +505,11 @@ metadata:
 spec:
   hostNetwork: true
   restartPolicy: OnFailure
+  tolerations:
```
Comment (alena1108) [inline rke/templates/calico.go]: This change could cause plugins redeploy on pre-existing clusters. Can you check if the spec update followed by pre-existing cluster modification  - any parameter that would trigger rke cluster reconcile - result in addon redeploy? @orangedeng

Comment (kinarashah) [reply rke/templates/calico.go]: I had the same question :) Discussed with @galal-hussein and yeah, the addons upgrades would happen

**Comment by alena1108** [comment]
Update: We will allow tolerations to all k8s versions, which will cause all system components to restart upon an edit of the cluster that triggers a RKE up. This fact will be release noted for v2.3.

**Comment by StrongMonkey** [comment]
I have a question regarding to this change. If we are allowing all the taints, will the pod get scheduled into unhealthy node? How are we preventing that? @orangedeng Looks like this is causing https://github.com/rancher/rancher/issues/26190 and https://github.com/rancher/rancher/issues/26191.

**Pull request merged**